lib/core/api.js

```diff
@@ -171,7 +171,8 @@ api.upload.accounts = function(happyCb, sadCb) {
 };
 
 function postBlockToPlatform(data, groupId, callback) {
-  if (!callback) {
+
+  if (typeof groupId === 'function') {
     callback = groupId;
```

Comment:
Yeah, I think this is where you commented on this before. Not quite sure why it's in the diff, but I'll investigate. I think it was something we put on hold to change because it will require a change to platform-client? Or am I misremembering?

I agree that the hidden optionality is confusing.